src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 10 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter.Get-ChildItem -Path -Filter "file*" -Hidden
```powershell
Expected 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 379
379:             $result.Count | Should -Be 1
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter.Get-ChildItem -Path -Filter "file*" -Force
```powershell
Expected 3, but got 2.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 386
386:             $result.Count | Should -Be 3
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter -Recurse.Get-ChildItem -Path $rootDir -Filter "file*" -Hidden -Recurse
```powershell
Expected 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 426
426:             $result.Count | Should -Be 3
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter -Recurse.Get-ChildItem -Path $rootDir -Filter "file*" -Force -Recurse
```powershell
Expected 10, but got 7.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 435
435:             $result.Count | Should -Be 10
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Filter -Recurse -Name.Get-ChildItem -Path -Filter "file*" -Hidden -Recurse -Name
```powershell
Expected 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystemProviderExtended.Tests.ps1: line 470
470:             $result.Count | Should -Be 3
```